djautotask/api_rest.py

```diff
@@ -206,6 +206,13 @@ def build_api_base_url(self):
             self.API,
         )
 
+    def get_api_base_url(self, endpoint):
```

Comment:
@kti-daniel I didn't see a response from you about this. What do you think about it?

Reply:
Oh, what happened? I missed this comment and the comment below. I will look into those. 

It seemed the email notice was not working temporarily last week. Weird @@

Reply:
using parameter such as `endpoint=self.API` is not possible; one sets up `self.api_base_url`, and the other needs to return the built endpoint. So I think the following is possible. How about that?

```
    def build_api_base_url(self):
        self.api_base_url = self.get_api_base_url()

    def get_api_base_url(self, endpoint=None):
        if not endpoint:
            endpoint = self.API
            
        return '{0}v{1}/{2}/'.format(
            self.server_url,
            settings.AUTOTASK_CREDENTIALS['rest_api_version'],
            endpoint,
        )
```

Reply:
My mistake! I didn't realize we couldn't use API as a default arg. 

Your solution looks good! I think it could be simplified even further though. Something like:
```python
    def build_api_base_url(self, endpoint=None):

        if not endpoint:
            endpoint = self.API

        return '{0}v{1}/{2}/'.format(
            self.server_url,
            settings.AUTOTASK_CREDENTIALS['rest_api_version'],
            endpoint,
        )
```
Then we only need one method. Should be a small change. I can make it if you don't have any objections.

Reply:
I am afraid `self.api_base_url ` are not set up in the above method. Can you verify that?

Reply:
Forgot to mention that. No it's not set in `build_api_base_url`. Instead I put ` self.api_base_url = self.build_api_base_url()` in the API client's `__ini__`. 

It seemed better to me to only concern the `build_api_base_url` method with building the URL. Not building it AND setting it on the client. Plus, my understanding is that  `__init__` is a good place to set initial properties for our object.

Reply:
That sounds good! Let's do that. I have no objection.

Reply:
I found this is not pushed yet. I will remember that. And the next time to work REST API, I will improve that. Thanks Cameron.

Reply:
Sorry about that Daniel. Sounds good. I don't know how I missed pushing to that branch :facepalm: